Core/Tests/Program.cs

```diff
@@ -23,122 +23,143 @@ namespace Loyc.Tests
 {
 	public class RunCoreTests
 	{
-		public static readonly VList<Pair<string, Action>> Menu = new VList<Pair<string, Action>>()
+		public static readonly VList<Pair<string, Func<int>>> Menu = new VList<Pair<string, Func<int>>>()
 		{
-			new Pair<string,Action>("Run unit tests of Loyc.Essentials.dll",  Loyc_Essentials),
-			new Pair<string,Action>("Run unit tests of Loyc.Collections.dll", Loyc_Collections),
-			new Pair<string,Action>("Run unit tests of Loyc.Syntax.dll",      Loyc_Syntax),
-			new Pair<string,Action>("Run unit tests of Loyc.Utilities.dll",   Loyc_Utilities),
+			new Pair<string,Func<int>>("Run unit tests of Loyc.Essentials.dll",  Loyc_Essentials),
+			new Pair<string,Func<int>>("Run unit tests of Loyc.Collections.dll", Loyc_Collections),
+			new Pair<string,Func<int>>("Run unit tests of Loyc.Syntax.dll",	  Loyc_Syntax),
+			new Pair<string,Func<int>>("Run unit tests of Loyc.Utilities.dll",   Loyc_Utilities),
 		};
 
 		public static void Main(string[] args)
 		{
 			// Workaround for MS bug: Assert(false) will not fire in debugger
 			Debug.Listeners.Clear();
 			Debug.Listeners.Add( new DefaultTraceListener() );
-			RunMenu(Menu);
+			if (RunMenu(Menu, args.Length > 0 ? args[0].GetEnumerator() : null) > 0)
+				// Let the outside world know that something
```

Comment:
Looks like you have a habit of really short comment lines. Please don't break before col 80 (I break at 85 - most people go higher)